### ISSUE TITLE
Document how to use zsh

### ISSUE THREAD

**Issue by lesteve** [jean-zay-users/jean-zay-doc#18]
- install zsh from conda-forge
- `source /gpfslocalsup/spack_soft/environment-modules/4.3.1/gcc-4.8.5-ism7cdy4xverxywj27jvjstqwk5oxe2v/init/zsh` so that the `module` command works

**Comment by mypey** [comment]
Hi @lesteve,
just one comment here: the official shell on Jean Zay is _bash_. In particular, all system scripts are bash scripts. Additional shells are installed but we do not provide support for them yet. _zsh_ might work very well (and I hope it will) but I just want to warn you that if there are some troubles when using it, this will be considered as a low priority issue.

**Comment by lesteve** [comment]
Thanks a lot for the precision! I think I have seen it somewhere in your official doc that bash was the only supported shell indeed. I will make sure to mention that `zsh` is not supported officially!

In practice I have not noticed any issue so far so I thought I would mention that if people are interested. 

Just to explain, I am used to have a `zsh` prompt and there are a few `zsh` features or rather [prezto](https://github.com/sorin-ionescu/prezto) (a zsh framework similar to oh-my-zsh) that I can not live without (nice prompt, autosuggestions, Control-P matching anywhere in my history, Alt-P matching history with the whole beginning of my command, etc ...).

Basically this is a way to feel more at home on Jean Zay.

**Comment by rstrudel** [comment]
It does work for fish too. I have a similar setup:
- install it from conda-forge
- to get the module command
`source /gpfslocalsup/spack_soft/environment-modules/4.3.1/gcc-4.8.5-ism7cdy4xverxywj27jvjstqwk5oxe2v/init/fish`
- add `exec $PATH_TO_CONDA_ENVS/bin/fish -l` to your `.bashrc`

**Comment by lesteve** [comment]
Excellent thanks for letting us know!

**Comment by RemiLacroix-IDRIS** [comment]
>     * `source /.../environment-modules/4.3.1/gcc-4.8.5-ism7cdy4xverxywj27jvjstqwk5oxe2v/init/zsh` so that the `module` command works. Probably needs something a bit more robust than this complicated name.

It would indeed to better to use `/.../environment-modules/current/init/...` instead.

**Comment by lesteve** [comment]
Yep I looked at my `.zshrc` and that's what I have indeed (I'll edit my first post):
```
MODULE_INIT_ZSH=/gpfslocalsup/spack_soft/environment-modules/current/init/zsh
if [[ -s "$MODULE_INIT_ZSH" ]]; then
  source "$MODULE_INIT_ZSH"
fi
```

**Comment by RemiLacroix-IDRIS** [comment]
In fact we even have an environment variable `$MODULESHOME` but I am not sure if it's inherited by zsh?

**Comment by BelCorentin** [comment]
In case you plan to use oh-my-zsh, it works, you just need to do:

`export PATH=/linkhome/rech/genmfd01/ulg98mt/bin:$PATH`

before running 

`sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"`

_Note:_ 
To get the previous code working, since exec zsh wouldn't work, I had to do 

```
if [ -x "$HOME/bin/zsh" ]; then
    export SHELL="$HOME/bin/zsh"
    exec "$HOME/bin/zsh"
else
    echo "zsh not found, falling back to bash."
fi
```

Which is actually safer (no risk about breaking your session / connection) (the hacky solution is pretty hard to do!)

**Comment by BelCorentin** [comment]
In case you plan to use oh-my-zsh, it works, you just need to do:

export PATH=$HOME/bin:$PATH

before running

sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"

Note:
To get the previous code working, since exec zsh wouldn't work, I had to do

if [ -x "$HOME/bin/zsh" ]; then
    export SHELL="$HOME/bin/zsh"
    exec "$HOME/bin/zsh"
else
    echo "zsh not found, falling back to bash."
fi

Which is actually safer (no risk about breaking your session / connection) (the hacky solution is pretty hard to do!)

**Comment by clementaudic** [comment]
Little modifications to points 1 and 3 of @flbbb's answer:

* To avoid `bash: exec: zsh: cannot execute: Is a directory`, name the downloaded directory something different than `zsh`, for example `zsh_source_code`.
    ```
    wget -O zsh.tar.xz https://sourceforge.net/projects/zsh/files/latest/download
    mkdir zsh_source_code && unxz zsh.tar.xz && tar -xvf zsh.tar -C zsh_source_code --strip-components 1
    cd zsh_source_code
    ./configure --prefix=$HOME
    make
    make install && cd .. && rm -rf zsh_source_code zsh.tar
    ```
* Add the following to `~/.bashrc`:
    ```
    exec ~/bin/zsh
    ```
    or equivalently:
    ```
    export PATH="$HOME/bin:$PATH"
    exec zsh
   ```
    Beware that `~/.bashrc` is by default not sourced on login on Jean-Zay; follow [these instructions to source it from ~/.bash_profile](https://jean-zay-doc.readthedocs.io/en/latest/tips/shell/#bash-config).
